graylog2-web-interface/src/components/common/Scratchpad.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useContext, useState, useEffect, useRef } from 'react';
+import styled, { css } from 'styled-components';
+import { rgba } from 'polished';
+import ClipboardJS from 'clipboard';
+
+import teinte from 'theme/teinte';
+import Interactable from 'components/common/Interactable';
+import { Alert, Button, MenuItem, SplitButton } from 'components/graylog';
+import { BootstrapModalConfirm } from 'components/bootstrap';
+import Icon from 'components/common/Icon';
+import { ScratchpadContext } from 'routing/context/ScratchpadProvider';
+
+const LOCALSTORAGE_ITEM = 'gl-scratchpad';
```

Comment:
Do we also want to make this preference user-dependent as we discussed at some point in the meeting week?
In case we want to do that later, we should clear the local storage key when the user logs out, just to be sure no information can be read by someone else, even if we warn no important information should be written in the scratchpad.